nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -686,6 +688,77 @@ impl DataStore {
         Ok(())
     }
 
+    /// Returns information about access for all of the given Nexus ids
+    ///
+    /// This set is assumed to be pretty small.
+    pub async fn database_nexus_access_all(
+        &self,
+        opctx: &OpContext,
+        nexus_ids: &BTreeSet<OmicronZoneUuid>,
+    ) -> Result<Vec<DbMetadataNexus>, Error> {
+        use nexus_db_schema::schema::db_metadata_nexus::dsl;
+
+        let db_nexus_ids: BTreeSet<_> = nexus_ids
+            .iter()
+            .cloned()
```

Comment:
Nit - since UUIDs impl `Copy`, I think this could be

```suggestion
            .copied()
```

(which is not different in terms of generated code but it's more obvious that this is cheap)

Reply:
Fixed in e8eb204f96309d21e84e43feee2b09ad635d7ddd.